actuator/src/main/java/org/tron/core/actuator/WithdrawBalanceToActuator.java

```diff
@@ -0,0 +1,171 @@
+    ret.setWithdrawAmount(allowance);
+    ret.setStatus(fee, code.SUCESS);
+
+    return true;
+  }
+
+  @Override
+  public boolean validate() throws ContractValidateException {
+    if (this.any == null) {
+      throw new ContractValidateException(ActuatorConstant.CONTRACT_NOT_EXIST);
+    }
+    if (chainBaseManager == null) {
+      throw new ContractValidateException(ActuatorConstant.STORE_NOT_EXIST);
+    }
+    AccountStore accountStore = chainBaseManager.getAccountStore();
+    DynamicPropertiesStore dynamicStore = chainBaseManager.getDynamicPropertiesStore();
+    MortgageService mortgageService = chainBaseManager.getMortgageService();
+    if (!this.any.is(WithdrawBalanceToContract.class)) {
+      throw new ContractValidateException(
+          "contract type error, expected type [WithdrawBalanceToContract], real type[" + any
+              .getClass() + "]");
+    }
+    final WithdrawBalanceToContract withdrawBalanceToContract;
+    try {
+      withdrawBalanceToContract = this.any.unpack(WithdrawBalanceToContract.class);
+    } catch (InvalidProtocolBufferException e) {
+      logger.debug(e.getMessage(), e);
+      throw new ContractValidateException(e.getMessage());
+    }
+    byte[] ownerAddress = withdrawBalanceToContract.getOwnerAddress().toByteArray();
+    if (!DecodeUtil.addressValid(ownerAddress)) {
+      throw new ContractValidateException("Invalid owner address");
+    }
+
+    AccountCapsule accountCapsule = accountStore.get(ownerAddress);
+    String readableOwnerAddress = StringUtil.createReadableString(ownerAddress);
+    if (accountCapsule == null) {
+      throw new ContractValidateException(
+          ACCOUNT_EXCEPTION_STR + readableOwnerAddress + NOT_EXIST_STR);
+    }
+
+    byte[] receiverAddress = withdrawBalanceToContract.getReceiverAddress().toByteArray();
+    if (!DecodeUtil.addressValid(receiverAddress)) {
+      throw new ContractValidateException("Invalid receiver address");
+    }
+
+    AccountCapsule receiverCapsule = accountStore.get(receiverAddress);
+    if (receiverCapsule == null) {
+      String readableReceiverAddress = StringUtil.createReadableString(receiverAddress);
+      throw new ContractValidateException(
+          ACCOUNT_EXCEPTION_STR + readableReceiverAddress + NOT_EXIST_STR);
+    }
+
+    boolean isGP = CommonParameter.getInstance()
+        .getGenesisBlock().getWitnesses().stream().anyMatch(witness ->
+            Arrays.equals(ownerAddress, witness.getAddress()));
+    if (isGP) {
+      throw new ContractValidateException(
+          ACCOUNT_EXCEPTION_STR + readableOwnerAddress
+              + "] is a guard representative and is not allowed to withdraw Balance");
+    }
+
+    long latestWithdrawTime = accountCapsule.getLatestWithdrawTime();
+    long now = dynamicStore.getLatestBlockHeaderTimestamp();
+    long witnessAllowanceFrozenTime = dynamicStore.getWitnessAllowanceFrozenTime() * FROZEN_PERIOD;
+
+    if (now - latestWithdrawTime < witnessAllowanceFrozenTime) {
```

Comment:
Why should we add the frozen time limit?

Reply:
Please refer to the historical code implementation.

Reply:
Disculpa y eso dónde se consulta